app/scripts/components/exploration/types.d.ts.ts

```diff
@@ -27,9 +35,15 @@ export interface TimelineDatasetAnalysis {
   };
 }
 
+export interface TimelineDatasetData extends DatasetLayer {
+  isPeriodic: boolean;
+  timeDensity: TimeDensity;
+  domain: Date[];
+}
+
 export interface TimelineDataset {
   status: TimelineDatasetStatus;
-  data: any;
+  data: TimelineDatasetData;
   error: any;
   settings: {
     // user defined settings like visibility, opacity
```

Comment:
@nerik How much type narrowing should we be doing?

The `isPeriodic` and so on properties are only available when the status is `success`, otherwise they're `undefined`. We could do this type narrowing or set it to be possibly undefined. On one hand we have a more complicated type, but with `undefined`, we need to check every prop, instead of relying on status. Thoughts?